radix-engine/src/system/system_callback.rs

```diff
@@ -47,11 +52,29 @@ use radix_transactions::model::{Executable, PreAllocatedAddress};
 
 pub const BOOT_LOADER_SYSTEM_SUBSTATE_FIELD_KEY: FieldKey = 1u8;
 
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+pub struct SystemParameters {
+    pub network_definition: NetworkDefinition,
+    pub costing_parameters: CostingParameters,
+    pub limit_parameters: LimitParameters,
+    pub max_per_function_royalty_in_xrd: Decimal,
+}
+
+impl Default for SystemParameters {
+    fn default() -> Self {
+        Self {
+            network_definition: NetworkDefinition::mainnet(),
+            costing_parameters: CostingParameters::default(),
+            limit_parameters: LimitParameters::default(),
+            max_per_function_royalty_in_xrd: Decimal::try_from(MAX_PER_FUNCTION_ROYALTY_IN_XRD)
+                .unwrap(),
+        }
+    }
+}
+
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
 pub enum SystemBoot {
-    V1 {
-        costing_parameters: CostingParameters,
-    },
+    V1(SystemParameters),
```

Comment:
Actually I'd ask we remove this structure for historical reason.

For `anenome`, we have been using 
```
VMBoot {
     V1 {
         scrypto_vm_version: u64
     }
}
```

Reply:
I'm with Josh.

In general, I like that we have structs for each enum variant of a versioned enum. It makes it easier to reason about, and you can have methods which only take a certain version.

At some point I have been wanting to add an `#[sbor(transparent)]` attribute for enum variants, so that we don't pay the overhead of a wrapped tuple during SBOR encode / decode -- but I haven't got round to it yet.

It's one of the few reasons the transaction stuff is quite manual, to hand-roll this optimization.